File: frontend/src/models/Orphanage.ts
import { OrphanageResponse } from "@/dtos/Orphanage";

class Orphanage {
  public constructor(
    public readonly id: number,
    public readonly name: string,
    public readonly latitude: number,
    public readonly longitude: number,
    public readonly about: string,
    public readonly instructions: string,
    public readonly openingHours: string,
    public readonly openOnWeekends: boolean,
    public readonly images: string[]
  ) {}

  public static fromResponse(response: OrphanageResponse) {
    return new Orphanage(
      response.id,
      response.name,
      response.latitude,
      response.longitude,
      response.about,
      response.instructions,
      response.opening_hours,
      response.open_on_weekends,
      response.images
    );
  }
}

export default Orphanage;

File: mobile/lib/dtos/orphanage.dart
import 'dart:io';

class OrphanageCreateData {
  final String name;
  final num latitude;
  final num longitude;
  final String about;
  final String instructions;
  final String openingHours;
  final bool openOnWeekends;
  final List<File> images;

  const OrphanageCreateData({
    this.name,
    this.latitude,
    this.longitude,
    this.about,
    this.instructions,
    this.openingHours,
    this.openOnWeekends,
    this.images,
  });
}
